cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/batch/sink/DBSink.java

```diff
@@ -140,8 +143,9 @@ private void setResultSetMetadata() throws Exception {
            ResultSet rs = statement.executeQuery(String.format("SELECT %s FROM %s WHERE 1 = 0",
                                                                dbSinkConfig.columns, dbSinkConfig.tableName))
       ) {
+        List<String> columns = ImmutableList.copyOf(Splitter.on(",").split(dbSinkConfig.columns));
         ResultSetMetaData resultSetMetadata = rs.getMetaData();
-        int columnCount = resultSetMetadata.getColumnCount();
+        int columnCount = columns.size();
```

Comment:
@gokulavasan mentioned the test case that raised this issue after I opened the PR. After testing, I discovered this issue and it's now fixed.